services/authoring/src/repositories/git-repository.js

```diff
@@ -93,24 +103,22 @@ class GitRepository {
     const history = uniqSort(historyEntries);
 
     if (history.length === 0) {
-      const commit = await this._repo.getCommit(sha);
+      const commit = await (revision ? this._repo.getCommit(revision) : this._repo.getMasterCommit());
       const tree = await commit.getTree();
 
-      const exists = await reduce(
-        fileNames,
-        async (exists, fileName) => {
-          if (exists) return true;
-          try {
-            await tree.getEntry(fileName);
-            return true;
-          } catch (err) {
-            return false;
-          }
-        },
-        false,
-      );
-
-      if (exists) return [mapEntry({ commit })];
+      for (let i = 0; i < fileNames.length; i++) {
+        try {
+          const fileName = fileNames[i];
+          await tree.getEntry(fileName);
+
+          return [{
+            sha: commit.sha(),
+            author: commit.author().name(),
+            date: commit.date(),
+            message: commit.message(),
+          }];
+        } catch (err) {}
```

Comment:
If at least one file is in the tree, i return the latest commit, if it isn't in the tree an error is thrown, and i ignore it